ocaml/lib/snarky_js_bindings_lib.ml

```diff
@@ -1155,18 +1155,31 @@ type sponge =
   | Checked of Poseidon_sponge_checked.t
   | Unchecked of Poseidon_sponge.t
 
+let hash_array (xs : field_class Js.t Js.js_array Js.t) (is_checked : bool Js.t)
+    =
+  let input = Array.map (Js.to_array xs) ~f:of_js_field in
+  if Js.to_bool is_checked then Random_oracle.Checked.hash input
+  else Random_oracle.hash (Array.map ~f:to_unchecked input) |> Field.constant
+
+
 let poseidon =
   object%js
     (* this could be removed eventually since it's easily implemented using `update` *)
     method hash (xs : field_class Js.t Js.js_array Js.t)
         (is_checked : bool Js.t) : field_class Js.t =
-      let input = Array.map (Js.to_array xs) ~f:of_js_field in
+      to_js_field (hash_array xs is_checked)
+
+    method hashToGroup (xs : field_class Js.t Js.js_array Js.t)
+        (is_checked : bool Js.t) =
+      let input = hash_array xs is_checked in
       let digest =
-        if Js.to_bool is_checked then Random_oracle.Checked.hash input
+        if Js.to_bool is_checked then
+          Snark_params.Group_map.Checked.to_group input
         else
-          Random_oracle.hash (Array.map ~f:to_unchecked input) |> Field.constant
+          let x, y = Snark_params.Group_map.to_group (to_unchecked input) in
+          (Field.constant @@ x, Field.constant @@ y)
       in
-      to_js_field digest
+      to_js_group (fst digest) (snd digest)
```

Comment:
```
if Js.to_bool is_checked then Snark_params.Group_map.Checked.to_group input
else
  let x, y = Snark_params.Group_map.to_group (to_unchecked input) in
  to_js_group (Field.constant x) (Field.constant y)
```
- the `@@` is unneeded
- don't need to construct `digest` to immediately destruct it through `fst` and `snd`